src/DCSAM.cpp

```diff
@@ -30,8 +30,18 @@ void DCSAM::update(const gtsam::NonlinearFactorGraph &graph,
                    const gtsam::DiscreteFactorGraph &dfg,
                    const DCFactorGraph &dcfg,
                    const gtsam::Values &initialGuessContinuous,
-                   const DiscreteValues &initialGuessDiscrete) {
-  // First things first: combine currContinuous_ estimate with the new values
+                   const DiscreteValues &initialGuessDiscrete,
+                   const gtsam::FactorIndices &removeFactorIndices,
+                   const std::vector<size_t> &removeDiscreteFactorIndices) {
+
+  // First things first: get rid of factors that are to be removed so updates
+  // to follow take the removals into account 
+  isam_.update(gtsam::NonlinearFactorGraph(), gtsam::Values(), removeFactorIndices);
```

Comment:
Maybe let's see if we can combine this with the `isam_.update` after the discrete update?